master/internal/task/allocation.go

```diff
@@ -1093,63 +1122,90 @@ func (a *allocation) terminated(reason string) {
 	case a.killedWhileRunning:
 		exitReason = fmt.Sprintf("allocation killed after %s", reason)
 		a.syslog.Info(exitReason)
+		a.SetExitStatus(exitReason, a.getErrCode())
```

Comment:
setting exit status here and in all the different code paths feels easy to forgot to add it for a path 

I wonder if we could just fields on allocation then have a cleanup function be called? I don't really know if this would be better, but it feels to me like we should just end the allocation then persist the exit reason / status if we have it in one place

thoughts @stoksc ?

Reply:
Commented elsewhere, it really only needs to be called in 2 places. One of (resources state changed path or terminated) and the cleanup func (basically a finally block)

Reply:
Let's chat tomorrow about this.